docs/developers/performance-test-setup-for-karmada.md

```diff
@@ -0,0 +1,331 @@
+
+- Create 10 namespace
+
+- Create 20 deployments with 1000 pods inside that namespace
+
+
+We will create file `config.yaml` that describes this test. First we need to start with defining test name:
+
+```yaml
+name: test
+```
+
+ClusterLoader2 will create namespaces automatically, but we need to specify how many namespaces we want:
+
+```yaml
+namespace:
+  number: 10
+```
+
+Next, we need to specify TuningSets. TuningSet describes how actions are executed. The qps means 1/qps s per action interval. In order to distribute resources slowly to relieve the pressure on the apiserver, the qps of Uniformtinyqps is set to 0.01, which means that after distributing a deployment, we wait 10s before continuing to distribute the next deployment.
+
+```yaml
+tuningSets:
+- name: Uniformtinyqps
+    qpsLoad:
+    qps: 0.01
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+```
+
+Finally, we will create a phase that creates deployment and propagation policy . We need to specify in which namespaces we want the deployment and propagation policy to be created, how many of these deployments per namespace. Also, we will need to specify template for our deployment and propagation policy , which we will do later. For now, let's assume that this template allows us to specify number of replicas in deployment.
+
+```yaml
+steps:
+- name: Create deployment
+  phases:
+  - namespaceRange:
+      min: 1
+      max: 10
+    replicasPerNamespace: 20
+    tuningSet: Uniformtinyqps
+    objectBundle:
+    - basename: test-deployment
+      objectTemplatePath: "deployment.yaml"
+      templateFillMap:
+        Replicas: 1000
+  - namespaceRange:
+      min: 1
+      max: 10
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    - basename: test-pp
+      objectTemplatePath: "pp.yaml"
+      templateFillMap:
+        Replicas: 1
+        
+```
+
+Whole `config.yaml` will look like this:
+
+```yaml
+name: test
+
+namespace:
+  number: 10
+
+tuningSets:
+- name: Uniformtinyqps
+    qpsLoad:
+    qps: 0.1
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+
+steps:
+- name: Create deployment 
+  phases:
+  - namespaceRange:
+      min: 1
+      max: 10
+    replicasPerNamespace: 20
+    tuningSet: Uniformtinyqps
+    objectBundle:
+    - basename: test-deployment
+      objectTemplatePath: "deployment.yaml"
+      templateFillMap:
+        Replicas: 1000
+  - namespaceRange:
+      min: 1
+      max: 10
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    objectBundle:
+    - basename: test-pp
+      objectTemplatePath: "pp.yaml"
+```
+
+Now, we need to specify deployment and propagation template. ClusterLoader2 by default adds parameter `Name` that you can use in your template. In our config, we also passed `Replicas` parameter. We need to remember to set correct labels so PodStartupLatency and WaitForControlledPodsRunning will watch correct pods. So our template for deployment will look like this (`deployment.yaml` file):
```

Comment:
set correct labels so PodStartupLatency and WaitForControlledPodsRunning will watch correct pods? It seems we don't test about PodStartupLatency.